api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.BlobCertifiedTimeout)
+	defer cancel()
+	blobStatusReply, err := pd.pollBlobStatus(timeoutCtx, blobKey, blobStatus.ToProfobuf())
+	if err != nil {
+		return nil, fmt.Errorf("poll blob status: %w", err)
+	}
+	pd.logger.Debug("Blob status CERTIFIED", "blobKey", blobKey)
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.ContractCallTimeout)
+	defer cancel()
+	nonSignerStakesAndSignature, err := pd.certVerifier.GetNonSignerStakesAndSignature(
+		timeoutCtx, blobStatusReply.GetSignedBatch())
+	if err != nil {
+		return nil, fmt.Errorf("get non signer stake and signature: %w", err)
+	}
+	pd.logger.Debug("Retrieved NonSignerStakesAndSignature", "blobKey", blobKey)
+
+	eigenDACert, err := verification.BuildEigenDACert(
+		blobStatusReply.GetBlobInclusionInfo(),
+		blobStatusReply.GetSignedBatch().GetHeader(),
+		nonSignerStakesAndSignature)
+	if err != nil {
+		return nil, fmt.Errorf("build eigen da cert: %w", err)
+	}
+	pd.logger.Debug("Constructed EigenDACert", "blobKey", blobKey)
+
+	timeoutCtx, cancel = context.WithTimeout(ctx, pd.config.ContractCallTimeout)
+	defer cancel()
+	err = pd.certVerifier.VerifyCertV2(timeoutCtx, eigenDACert)
+	if err != nil {
+		return nil, fmt.Errorf("verify cert for blobKey %v: %w", blobKey, err)
+	}
+	pd.logger.Debug("EigenDACert verified", "blobKey", blobKey)
+
+	return eigenDACert, nil
+}
+
+// Close is responsible for calling close on all internal clients. This method will do its best to close all internal
+// clients, even if some closes fail.
+//
+// Any and all errors returned from closing internal clients will be joined and returned.
+//
+// This method should only be called once.
+func (pd *PayloadDisperser) Close() error {
+	err := pd.disperserClient.Close()
+	if err != nil {
+		return fmt.Errorf("close disperser client: %w", err)
+	}
+
+	return nil
+}
+
+// pollBlobStatus polls the disperser for the status of a blob that has been dispersed
+//
+// This method will only return a non-nil BlobStatusReply if the blob is reported to be CERTIFIED prior to the timeout.
+// In all other cases, this method will return a nil BlobStatusReply, along with an error describing the failure.
+func (pd *PayloadDisperser) pollBlobStatus(
+	ctx context.Context,
+	blobKey core.BlobKey,
+	initialStatus dispgrpc.BlobStatus,
+) (*dispgrpc.BlobStatusReply, error) {
+
+	previousStatus := initialStatus
+
+	ticker := time.NewTicker(pd.config.BlobStatusPollInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf(
+				"timed out waiting for %v blob status, final status was %v: %w",
+				dispgrpc.BlobStatus_CERTIFIED,
+				previousStatus,
+				ctx.Err())
+		case <-ticker.C:
+			// This call to the disperser doesn't have a dedicated timeout configured.
+			// If this call fails to return in a timely fashion, the timeout configured for the poll loop will trigger
```

Comment:
is this actually true?

Reply:
There is a timeout set on the context being passed into `pollBlobStatusUntilCertified`.

If that timeout triggers, it will abort this `GetBlobStatus` if necessary, *and* abort the polling loop, right?